### ISSUE TITLE
test: add unit tests for sequence-level KL divergence

### ISSUE THREAD

**Issue by RyderFreeman4Logos** [RyderFreeman4Logos/heretic#21]
## Problem

The sequence-level KL feature (PR #17) has no dedicated unit tests. Key logic that should be tested:

1. **Logits extraction offset** — `get_sequence_logprobs()` extracts `logits[:, prompt_len-1 : prompt_len+seq_len-1, :]`. An off-by-one here silently produces wrong KL values.

2. **Padding mask correctness** — `reference_mask` must be `1` for real tokens, `0` for padding. Incorrect masking dilutes or inflates KL.

3. **KL formula** — `F.kl_div(logprobs, base_logprobs, reduction="none", log_target=True)` should match the expected mathematical definition.

4. **Edge cases**:
   - Batch with mixed-length sequences (some padded, some not)
   - Tokenizer without `pad_token_id` (fallback to `eos_token_id`)
   - `kl_sequence_length` longer than actual generated response

## Suggested Test Cases

```python
def test_sequence_logprobs_offset():
    """Verify logits are extracted at correct positions for teacher forcing."""

def test_padding_mask_excludes_pad_tokens():
    """KL should only average over non-padding positions."""

def test_kl_zero_when_distributions_identical():
    """Same model vs itself should yield KL ≈ 0."""

def test_pad_token_fallback():
    """Tokenizer without pad_token_id should fall back to eos_token_id."""
```

## Context

Found during review of PR #17 (`feat: add sequence-level KL divergence via teacher forcing`). The three iterations (a0721b9 → 08a7872 → eb37465) fixed real bugs (padding mask, O(n²) batch offset), confirming that this code path benefits from test coverage.
